src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,27 @@
 package mate.academy;
 
+import java.util.Random;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Random random = new Random();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        try {
+            Thread.sleep(random.nextInt(200));
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        return CompletableFuture.supplyAsync(() -> createUserData(userId), executor);
```

Comment:
After creating `UserData`, store the result in the cache to ensure that subsequent requests for the same `userId` can be served from the cache.